llama_index/evaluation/multi_modal/faithfulness.py

```diff
@@ -0,0 +1,222 @@
+
+    def evaluate(
+        self,
+        query: str | None = None,
+        response: str | None = None,
+        contexts: Sequence[str] | None = None,
+        image_paths: List[str] | None = None,
+        image_urls: List[str] | None = None,
+        **kwargs: Any,
+    ) -> EvaluationResult:
+        """Evaluate whether the response is faithful to the multi-modal contexts."""
+        del query  # Unused
+        del kwargs  # Unused
+        if contexts is None or response is None:
+            raise ValueError("contexts and response must be provided")
+
+        context_str = "\n\n".join(contexts)
+        fmt_prompt = self._eval_template.format(
+            context_str=context_str, query_str=response
+        )
+
+        image_documents = []
+        if image_paths:
+            for image_path in image_paths:
+                image_documents += SimpleDirectoryReader(
+                    input_files=[image_path]
+                ).load_data()
+            node_parser = SentenceSplitter.from_defaults()
+            image_nodes = node_parser.get_nodes_from_documents(image_documents)
+        if image_urls:
+            ...
+
+        response_obj = self._multi_modal_llm.complete(
+            prompt=fmt_prompt,
+            image_documents=image_nodes,
+        )
+
+        raw_response_txt = str(response_obj)
+
+        if "yes" in raw_response_txt.lower():
+            passing = True
+        else:
+            passing = False
+            if self._raise_error:
+                raise ValueError("The response is invalid")
+
+        return EvaluationResult(
+            response=response,
+            contexts=contexts,
+            passing=passing,
+            score=1.0 if passing else 0.0,
+            feedback=raw_response_txt,
+        )
+
+    async def aevaluate(
+        self,
+        query: str | None = None,
+        response: str | None = None,
+        contexts: Sequence[str] | None = None,
+        image_paths: List[str] | None = None,
+        image_urls: List[str] | None = None,
+        **kwargs: Any,
+    ) -> EvaluationResult:
+        """Async evaluate whether the response is faithful to the multi-modal contexts."""
+        del query  # Unused
+        del kwargs  # Unused
+        if contexts is None or response is None:
+            raise ValueError("contexts and response must be provided")
+
+        context_str = "\n\n".join(contexts)
+        fmt_prompt = self._eval_template.format(
+            context_str=context_str, query_str=response
+        )
+
+        image_documents = []
+        if image_paths:
+            for image_path in image_paths:
+                image_documents += SimpleDirectoryReader(
+                    input_files=[image_path]
+                ).load_data()
+            node_parser = SentenceSplitter.from_defaults()
+            image_nodes = node_parser.get_nodes_from_documents(image_documents)
+        if image_urls:
+            ...
```

Comment:
same notes as above

Reply:
we can probably share the node/document prep in a shared helper function between async and sync functions